Application/Items/Commands/Edit.cs

```diff
@@ -34,19 +34,20 @@ public Handler(ItemDbContext context, IMapper mapper)
             public async Task<Unit> Handle(Command request, CancellationToken cancellationToken)
             {
                 var item = await _context.Items.FindAsync(request.Id);
+                var user = await _context.Users.FindAsync(request.User);
 
                 if (item == null)
                 {
                     throw new NotFoundException(nameof(Item), request.Id);
                 }
 
-                if (item.Uploader != request.User)
+                if (item.Uploader.Username != user.Username)
                 {
                     throw new ConflictException($"Item {request.Id} does not belong to the client");
                 }
 
                 request.Item.Id = request.Id;
-                request.Item.Uploader = request.User;
+                request.Item.Uploader = user;
```

Comment:
check the value of request.item.uploader, because I think the wrong data is assigned